solidity/contracts/middleware/InterchainAccountRouter.sol

```diff
@@ -319,28 +345,74 @@ contract InterchainAccountRouter is
     }
 
     /**
-     * @notice Returns the local address of an interchain account
+     * @notice Returns the local address of a remotely owned interchain account
      * @dev This interchain account is not guaranteed to have been deployed
      * @param _origin The remote origin domain of the interchain account
-     * @param _router The remote origin InterchainAccountRouter
      * @param _owner The remote owner of the interchain account
+     * @param _router The remote InterchainAccountRouter
      * @param _ism The local address of the ISM
      * @return The local address of the interchain account
      */
     function getLocalInterchainAccount(
         uint32 _origin,
-        bytes32 _router,
         bytes32 _owner,
+        bytes32 _router,
         address _ism
     ) public view returns (OwnableMulticall) {
         return
             OwnableMulticall(
                 _getLocalInterchainAccount(
-                    _getSalt(_origin, _router, _owner, _ism)
+                    _getSalt(
+                        _origin,
+                        _owner,
+                        _router,
+                        TypeCasts.addressToBytes32(_ism)
+                    )
                 )
             );
     }
 
+    /**
+     * @notice Returns the remote address of a locally owned interchain account
+     * @dev This interchain account is not guaranteed to have been deployed
+     * @dev This function will only work if the destination domain is
+     * EVM compatible
+     * @param _owner The local owner of the interchain account
+     * @param _router The remote InterchainAccountRouter
+     * @param _ism The remote address of the ISM
+     * @return The local address of the interchain account
+     */
+    function getRemoteInterchainAccount(
+        address _owner,
+        address _router,
+        address _ism
+    ) public view returns (address) {
+        require(_router != address(0), "no router specified for destination");
+        bytes32 _salt = _getSalt(
+            localDomain,
+            TypeCasts.addressToBytes32(_owner),
+            TypeCasts.addressToBytes32(address(this)),
+            TypeCasts.addressToBytes32(_ism)
+        );
+        address _implementation = address(
+            uint160(
+                uint256(
+                    keccak256(
+                        abi.encodePacked(
+                            bytes1(0xd6),
+                            bytes1(0x94),
+                            _router,
+                            bytes1(0x01)
+                        )
+                    )
+                )
+            )
+        );
```

Comment:
maybe pull this into a CREATE library?

Reply:
or at least a comment about what this is

Reply:
Definitely deserves a comment